frontend-task/src/components/modal/modal.tsx

```diff
@@ -0,0 +1,41 @@
+import React from "react"
+import { Button, FormControlLabel, Modal, Radio, RadioGroup, Typography } from "@mui/material"
+
+type QuestionModalProps = {
+    errorMessage: string
+    handleSubmit: () => void
+    handleChange: (event: React.ChangeEvent<HTMLInputElement>) => void
+    isOpen: boolean
+}
+
+export const QuestionModal:React.FC<QuestionModalProps> = ({ errorMessage, handleSubmit, handleChange, isOpen}) => {
+    return(
+        <Modal
+        open={!isOpen}
+        style={{
+          display: "flex",
+          justifyContent: "center",
+          alignItems: "center",
+        }}
+      >
+        <div className="modal-content">
+          <Typography>3 + 2 = ?</Typography>
+          <RadioGroup
+            aria-labelledby="demo-radio-buttons-group-label"
+            defaultValue="3"
+            name="radio-buttons-group"
+            onChange={handleChange}
+          >
+            <FormControlLabel value="3" control={<Radio />} label="3" />
+            <FormControlLabel value="5" control={<Radio />} label="5" />
+            <FormControlLabel value="8" control={<Radio />} label="8" />
+            <FormControlLabel value="4" control={<Radio />} label="4" />
+          </RadioGroup>
+          {errorMessage && (
+            <Typography className="error-message">{errorMessage}</Typography>
```

Comment:
I think you need to move correct answer checking to this component rather then passing everything through the props